src/SFML/Window/Unix/VulkanImplX11.cpp

```diff
@@ -0,0 +1,228 @@
+//
+////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Window/Unix/VulkanImplX11.hpp>
+#include <SFML/Window/Unix/Display.hpp>
+#include <dlfcn.h>
+#define VK_USE_PLATFORM_XLIB_KHR
+#define VK_NO_PROTOTYPES
+#include <vulkan.h>
+#include <string>
+#include <map>
+#include <cstring>
+
+
+namespace
+{
+    // RAII Vulkan shared library wrapper
+    struct VulkanLibraryWrapper
+    {
+        // Ctor
+        VulkanLibraryWrapper() :
+        library(NULL)
+        {
+        }
+
+        // Dtor
+        ~VulkanLibraryWrapper()
+        {
+            if (library)
+                dlclose(library);
+        }
+
+        // Try to load the library and all the required entry points
+        bool loadLibrary()
+        {
+            if (library)
+                return true;
+
+            library = dlopen("libvulkan.so.1", RTLD_LAZY);
+
+            if (!library)
+                return false;
+
+            if (!loadEntryPoint(vkGetInstanceProcAddr, "vkGetInstanceProcAddr"))
+            {
+                dlclose(library);
+                library = NULL;
+                return false;
+            }
+
+            if (!loadEntryPoint(vkEnumerateInstanceLayerProperties, "vkEnumerateInstanceLayerProperties"))
+            {
+                dlclose(library);
+                library = NULL;
+                return false;
+            }
+
+            if (!loadEntryPoint(vkEnumerateInstanceExtensionProperties, "vkEnumerateInstanceExtensionProperties"))
+            {
+                dlclose(library);
+                library = NULL;
+                return false;
+            }
+
+            return true;
+        }
+
+        template<typename T>
+        bool loadEntryPoint(T& entryPoint, const char* name)
+        {
+            entryPoint = reinterpret_cast<T>(dlsym(library, name));
+
+            return (entryPoint != NULL);
+        }
+
+        void* library;
+
+        PFN_vkGetInstanceProcAddr vkGetInstanceProcAddr;
+        PFN_vkEnumerateInstanceLayerProperties vkEnumerateInstanceLayerProperties;
+        PFN_vkEnumerateInstanceExtensionProperties vkEnumerateInstanceExtensionProperties;
+    };
+
+    VulkanLibraryWrapper wrapper;
+}
+
+
+namespace sf
+{
+namespace priv
+{
+////////////////////////////////////////////////////////////
+bool VulkanImplX11::isAvailable(bool requireGraphics)
+{
+    static bool checked = false;
+    static bool computeAvailable = false;
+    static bool graphicsAvailable = false;
+
+    if (!checked)
+    {
+        checked = true;
+
+        // Check if the library is available
+        computeAvailable = wrapper.loadLibrary();
+
+        // To check for instance extensions we don't need to differentiate between graphics and compute
+        graphicsAvailable = computeAvailable;
+
+        if (graphicsAvailable)
```

Comment:
If we don't require graphics, do we still need to run this block? (See also required vs request etc above)

Reply:
Yes, because Vulkan doesn't differentiate between graphics and non-graphics extensions. We need to make that differentiation ourselves when checking for support. We enter the block regardless what we are looking for and update the available flags accordingly.